File: java-multi-threading/demo.threading09.workerdemo2/src/demo/threading09/workerdemo2/Basket.java
package demo.threading09.workerdemo2;

public class Basket {
	long items;

	
	public  void addItem(){
		
		//TODO: Add Lock
			long item=items;
			item++;
			items=item;
		
	}

	public long getItems() {
		return items;
	}

}

File: AsyncProgramming/Dem08/Basket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace Demo07
{
    public class Basket
    {
        public long Items { get; private set; }
        Random r = new Random();


        public void AddItem()
        {
            lock(this)  //try{Monitor.Enter(this)
            {
                var x = Items;
                x++;
                Items = x;
            }  // finally{ Montior.Exit(this) ;}
            
        }

        public void AddItemV2()
        {
            try
            {
                
                while(!Monitor.TryEnter(this))
                    Console.Write("!"); 
                var x = Items;
                x++;
                Items = x;
            }
            finally
            {
                Monitor.Exit(this);
            }
        }
    }
}
